core/server/adapters/scheduling/SchedulingDefault.js

```diff
@@ -197,61 +197,61 @@ SchedulingDefault.prototype._execute = function (jobs) {
  * - if we detect to publish a post in the past (case blog is down), we add a force flag
  */
 SchedulingDefault.prototype._pingUrl = function (object) {
-    debug('Ping url', object.url, moment().format('YYYY-MM-DD HH:mm:ss'), moment(object.time).format('YYYY-MM-DD HH:mm:ss'));
+    const {url, time} = object;
+
+    debug('Ping url', url, moment().format('YYYY-MM-DD HH:mm:ss'), moment(time).format('YYYY-MM-DD HH:mm:ss'));
 
     let timeout;
-    const {url, time} = object;
-    const httpMethod = object.extra ? object.extra.httpMethod : 'PUT',
-        tries = object.tries || 0,
-        requestTimeout = object.extra ? object.extra.timeoutInMS : 1000 * 5,
-        maxTries = 30,
-        req = request[httpMethod.toLowerCase()](url),
-        self = this;
+    const httpMethod = object.extra ? object.extra.httpMethod : 'PUT';
+    const tries = object.tries || 0;
+    const requestTimeout = object.extra ? object.extra.timeoutInMS : 1000 * 5;
+    const maxTries = 30;
+
+    const options = {
+        timeout: requestTimeout,
+        method: httpMethod.toLowerCase(),
+        json: true
+    };
 
     if (moment(time).isBefore(moment())) {
         if (httpMethod === 'GET') {
-            req.query('force=true');
+            // @todo: rename to searchParams when updating to Got v10
+            options.query = 'force=true';
         } else {
-            req.send({
-                force: true
-            });
+            options.body = {force: true};
         }
     }
 
-    req.timeout({
-        response: requestTimeout
-    });
-
-    req.end(function (err, response) {
-        if (err) {
-            // CASE: post/page was deleted already
-            if (response && response.status === 404) {
-                return;
-            }
-
-            // CASE: blog is in maintenance mode, retry
-            if (response && response.status === 503 && tries < maxTries) {
-                timeout = setTimeout(function pingAgain() {
-                    clearTimeout(timeout);
+    return request(url, options).catch((err) => {
+        const {statusCode} = err;
 
-                    object.tries = tries + 1;
-                    self._pingUrl(object);
-                }, self.retryTimeoutInMs);
+        // CASE: post/page was deleted already
+        if (statusCode === 404) {
+            return;
+        }
 
-                common.logging.error(new common.errors.GhostError({
-                    err,
-                    context: 'Retrying...',
-                    level: 'normal'
-                }));
+        // CASE: blog is in maintenance mode, retry
+        if (statusCode === 503 && tries < maxTries) {
+            timeout = setTimeout(() => {
+                clearTimeout(timeout);
```

Comment:
I don't think this is necessary? It was there when I got here 😄

Reply:
Yeah, doesn't seem like it's serving it's purpose. :fire: it :smile: